object.js

```diff
@@ -0,0 +1,209 @@
+// Keep track of number of dice input at the start & current game round
+var noOfDice;
+var trackPlayer = 0;
+
+/**
+ * Get dice rolls for curr player and populate current player's dice array
+ * Return the new dice rolls
+ */
+var rollDice = function () {
+  var randInt = Math.floor(Math.random() * 6) + 1;
+  return randInt;
+};
+
+/** Create Object to store game details of each player */
+class Player {
+  /**
+  * Tracks the initial dice roll, reordered dice & score
+  * @param {Array<Number>} initialDiceArray original dice number
+  * @param {Number} orderedDice store reordered dice number
+  * @param {Array<Number>} scoreArray store series of orderedDice for scoring
+  */
+  constructor(orderedDice) {
+    this.initialDiceArray = [];
+    this.orderedDice = orderedDice;
+    this.scoreArray = [];
+  }
+
+  /**
+  * Returns a dice array for player
+  * @param {String} numberOfDice user inputs number of dice to roll and store
+  */
+  generateDiceNumber(numberOfDice) {
+    for (var i = 0; i < numberOfDice; i += 1) {
+      this.initialDiceArray.push(rollDice());
+    }
+  }
+}
+
+/**
+ * Instantiate new Players & give them a unique name
+*/
+var createPlayers = function (numberOfPlayers) {
+  for (var i = 0; i < numberOfPlayers; i += 1) {
+    window['Player' + String(i + 1)] = new Player();
+    players.push(window['Player' + String(i + 1)]);
+  }
+};
+
+/**
+ *  Run a test to ensure input is number or not blank
+ *  Return a bollean valuue
+ */
+var passNumberTest = function (userInput) {
+  // Check: input is number, has correct length, or has appropriate order of indexes that match
+  if (isNaN(Number(userInput)) === true) return 'false';
+  if (userInput === '0' || userInput === '') return 'false';
+  return true;
+};
+
+// Create outside of this function so can be accessed by `passDiceOrderTest()` & `rearrangeDice()`
+var newPositions;
+/**
+* Return a boolean value that determines in input passes conditions
+* @param {String} userOrder user inputs string represent new order for dice
+*/
+var passDiceOrderTest = function (userOrder) {
+  // Create an array of indexes based on noOfDice
+  var positionOfDiceArr = Array.from({ length: noOfDice }, (v, k) => k);
+  // The input that user types is a string, we have to convert each one to Number
+  newPositions = userOrder.split('').map((i) => Number(i));
+  // Run check for every element of userOrder vs index of diceArr, ensuring they match
+  var inputCheckTest = positionOfDiceArr.every((i) => newPositions.includes(i));
+
+  // Check if input has appropriate order of indexes that match
+  if (inputCheckTest == false) {
+    return 'Uh oh! You entered an invalid order.';
+  }
+  return true;
+};
+
+/**
+* Returns a new dice number after new order has been keyed
+* @param {String} userOrder user inputs string represent new order for dice
+* @param {Number} current tracking number for current Player in players array
+*/
+var rearrangeDice = function (userOrder, current) {
```

Comment:
`current` is an ambiguous var name, would recommend `currentPlayerIndex` or something more verbose but clearer